### ISSUE TITLE
[bug] Exception `Error binding parameter 4 - probably unsupported type` on every failed to download file

### ISSUE THREAD

**Issue by GammaC0de** [pyload/pyload#1346]
Hi,
I get this exception on every failed file, no matter which hoster:
```
11.04.2015 19:16:29 WARNING   Download failed: 100Testfile | Parse error: Free download link not found | Plugin may be out of date
11.04.2015 19:16:30 INFO      UnSkipOnFail: Looking for skipped duplicates of: 100Testfile (pid:57)
11.04.2015 19:16:30 INFO      UnSkipOnFail: No duplicates found
Traceback (most recent call last):
  File "/share/pyload/module/database/DatabaseBackend.py", line 96, in processJob
    self.result = self.f(*self.args, **self.kwargs)
  File "/share/pyload/module/database/FileDatabase.py", line 767, in updateLink
    self.c.execute('UPDATE links SET url=?,name=?,size=?,status=?,error=?,package=? WHERE id=?', (f.url, f.name, f.size, f.status, f.error, str(f.packageid), str(f.id)))
InterfaceError: Error binding parameter 4 - probably unsupported type.
Database Error @ updateLink (PyFile 56: 100Testfile@TwoSharedCom,) {} Error binding parameter 4 - probably unsupported type.
11.04.2015 19:16:30 DEBUG     All downloads processed
```

**Comment by MeegoPy** [comment]
Hi,
I have the same error but I do not find the issue origin in plugins.
In [module/database/FileDatabase.py](https://github.com/pyload/pyload/blob/stable/module/database/FileDatabase.py#L766) __f.error__ is a __module.plugins.Plugin.Fail__ class. The request need __f.error.message__ which is the correct __str__ type to store.

I use type(error) in PyFile __init__() and type(f.error) in FileDatabase updateLink() 

```
28.04.2015 11:34:23 INFO      [PyFile]                   -> <type 'unicode'>
28.04.2015 11:34:23 INFO      Download starts: http://www.4shared.com/mp3/7_o0WEwDce/4shared.html
28.04.2015 11:34:23 WARNING   Download failed: http://www.4shared.com/mp3/7_o0WEwDce/4shared.html | Required account not found
28.04.2015 11:34:23 INFO      [FileDatabase]                           -> <type 'str'>
28.04.2015 11:34:23 INFO      UnSkipOnFail: Looking for skipped duplicates of: http://www.4shared.com/mp3/7_o0WEwDce/4shared.html (pid:13)
28.04.2015 11:34:23 INFO      [FileDatabase]                           -> <class 'module.plugins.Plugin.Fail'>
28.04.2015 11:34:23 INFO      UnSkipOnFail: No duplicates found
28.04.2015 11:34:23 INFO      [FileDatabase]                           -> <class 'module.plugins.Plugin.Fail'>
28.04.2015 11:34:23 DEBUG     All downloads processed
```

It looks like UnSkipOnFail change error type from __str__ (line 4) to __module.plugins.Plugin.Fail__ (line 6 and 8) but it's a supposition, it is the only file which change recently, but I do not understand what really happens ;)

**Comment by GammaC0de** [comment]
Ok,
I have found the solution for this: add `str()` to `f.error`.
Since its part of pyLoad's core, UpdateManager cannot fix it, so you have to patch it manually.
P.S.
I see this has already been fixed in 0.4.10

```patch
--- FileDatabase.py.bad 2011-12-29 14:06:01.000000000 +0200
+++ FileDatabase.py     2015-05-01 02:16:21.774318829 +0300
@@ -764,7 +764,7 @@

     @style.async
     def updateLink(self, f):
-        self.c.execute('UPDATE links SET url=?,name=?,size=?,status=?,error=?,package=? WHERE id=?', (f.url, f.name, f.size, f.status, f.error, str(f.packageid), str(f.id)))
+        self.c.execute('UPDATE links SET url=?,name=?,size=?,status=?,error=?,package=? WHERE id=?', (f.url, f.name, f.size, f.status, str(f.error), str(f.packageid), str(f.id)))

     @style.queue
     def updatePackage(self, p):

```
